### DISCUSSION TITLE
some bugs need to fix

### DISCUSSION THREAD

**Discussion by TiaNex-Com** [cycle/orm#342] [General]
1.  product has many productSkus

```
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1-kg' for key 'mt_product_sku.mt_product_sku_index_product_id_attribute_name_61a108014425b'   
(1-kg ,"1" is product id and "kg" is attribute name)

string(194) "INSERT INTO `mt_product_sku` (`sequence`, `status`, `created_at`, `updated_at`, `ip_address`, `product_id`, `attribute_name`, `weight`, `price`, `quantity`) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
```

the right action should be delete the same rows or ON DUPLICATE KEY UPDATE

2.Many To Many (product and category)
when add new category to product , error happend but remove
```
Transaction can't be finished. Some relations can't be resolved:
Create new `productCategory`
  - category.products:productCategory (Cycle\ORM\Relation\ShadowBelongsTo)
```

**Comment by roxblnfk** [comment]
1. ~~You should check your indexes. I think that index `mt_product_sku.mt_product_sku_index_product_id_attribute_name_61a108014425b` shouldn't be unique or should contain additional field~~
Oh i was wrong. Yes we need to fix this behavior. https://github.com/cycle/orm/issues/350

3. Should be fixed here: https://github.com/cycle/orm/releases/tag/v2.1.1
